### PR TITLE
⛔ feat(ban): Contract for banning attendees and IP/ranges

### PR DESCRIPTION
**Pull request by nickvergessen**
### 🚧 Tasks

- [x] Allows to send a ban
- [x] Get a list of bans
- [x] Remove a ban
- [x] Experience a ban
    - [x] If conversation is named `ban user`, the user with id `banned` is banned (can not join)
    - [x] If conversation is named `ban guest`, all guests are banned (can not join)
- [x] *forgot to do:* Feature capability: `ban-v1`

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/Controller/BanController.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	/**
+	 * Ban an actor or IP address
+	 *
+	 * Required capability: `ban-v1`
+	 *
+	 * @param string $token Conversation token
+	 * @param 'users'|'groups'|'circles'|'emails'|'federated_users'|'phones'|'ip' $actorType Type of actor to ban, or `ip` when banning a clients remote address
+	 * @psalm-param Attendee::ACTOR_*|'ip' $actorType Type of actor to ban, or `ip` when banning a clients remote address
+	 * @param string $actorId Actor ID or the IP address or range in case of type `ip`
+	 * @param string $internalNote Optional internal note
```
Comment (Antreesy) [inline lib/Controller/BanController.php:40]: image

File: lib/Controller/BanController.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+		return new DataResponse(
+			[
+				'id' => random_int(1, 1337),
+				'actorType' => $this->participant->getAttendee()->getActorType(),
+				'actorId' => $this->participant->getAttendee()->getActorId(),
+				'bannedType' => $actorType,
+				'bannedId' => $actorId,
+				'bannedTime' => time(),
```
Comment (Antreesy) [inline lib/Controller/BanController.php:63]: do we need later in addition expand it to banStartTime and banEndTime (in case for temporal bans)?

**Review by Antreesy — APPROVED** [review]
Overall fine, but I'm struggling now to imagine how it would be used later on (from moderator perspective)

File: lib/Controller/BanController.php (hunk 2 above)
Comment (nickvergessen) [reply lib/Controller/BanController.php:63]: We don't support temporary bans for now 🤷

File: lib/Controller/BanController.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	 * @param string $actorId Actor ID or the IP address or range in case of type `ip`
+	 * @param string $internalNote Optional internal note
+	 * @return DataResponse<Http::STATUS_OK, TalkBan, array{}>|DataResponse<Http::STATUS_BAD_REQUEST, array{error: string}, array{}>
+	 *
+	 * 200: Ban successfully
+	 * 400: Actor information is invalid
+	 */
+	#[PublicPage]
+	#[RequireModeratorParticipant]
+	public function banActor(string $token, string $actorType, string $actorId, string $internalNote): DataResponse {
```
Comment (nickvergessen) [inline lib/Controller/BanController.php:48]:
```suggestion
	public function banActor(string $token, string $actorType, string $actorId, string $internalNote = ''): DataResponse {
```

**Pull request merged**